### DISCUSSION TITLE
Responsive doesn't work

### DISCUSSION THREAD

**Discussion by fonnets** [bootstrap-email/bootstrap-email#130] [Help] [answered]
The responsive doesn't work, I tried a sample code (I bought the template package) https://app.bootstrapemail.com/templates/e_commerce-order.
On the online editor if I change the view in 'mobile' mode, the items are placed one under the other perfectly (with the correct rezising of images), but if I try to send the email with the test function, when I view it with ios gmail the columns do not adapt and they try to hug but the content is crushed, making the email unreadable.

Thanks

**Comment by stuyam** [comment]
Yeah that sounds plausible. Gmail has really poor responsive support. Grid is one of the use cases that could be much better supported in gmail. I think there is a big project there in improving grid support for gmail unfortunately. I will work on that thank you for bringing it up 👍

**Comment by stuyam** [comment]
I have been doing a lot more research and figured out the problem. I have posted in on stack overflow as I haven't been able to track down a fix for gmail. Responsive grids do work but other features can break the style blocks in gmail such as child descendant selectors `>`.  Posting here in case anyone has any insights or ideas into this tricky edge case: https://stackoverflow.com/questions/69441076/html-email-style-tags-removed-in-gmail-ios-app

**Comment by fonnets** [comment]
Hi, sorry if I'm only answering you now, but I missed the notification of your first reply. I realized that the responsive part doesn't seem to work with sending tests from the bootstrapemail server, while sending the same rendered code to the online editor with my Amazon ses (which should be the one you use too) works perfectly! So you probably have some extra formatting that you did before sending that doesn't allow the gmail ios app (I have no way to test android at the moment) to detect media queries. I even compacted all the code on one line to save bytes and even so it didn't break the media queries, the email remains perfect! Investigate your test email function.

I am satisfied with bootstrap email, I hope a php version arrives to install it directly on my server! Thanks
Federico

**Comment by stuyam** [comment]
Oh really? That is puzzling, yeah I am using Amazon SES through a ruby on rails server for the online editor. You are saying you sent the compiled code downloaded from https://app.bootstrapemail.com/templates/e_commerce-order and that worked? 

In the current example it was `max-w-150` that was including a specific style in the style head which was breaking gmail. So for example if you removed that class from your code and rerendering it probably worked. So just clarifying it was the exact same email.

Glad you are enjoying bootstrap email, Im working to improve it every day and I appreciate your feedback 🙏

**Comment by fonnets** [comment]
Yes, really! I didn't send exactly that code from my server. I built the email myself (the columns were the same as the e_commerce-order example) and converted it with your editor, sending the email with your test function, gmail ios seems to ignore the media queries, the exact same code sent with amazon ses from my server shows the items perfectly stacked on my gmail app. I tested again last night, I confirm that it only depends on sending tests from your server, it does not depend on the code of the template. You need to investigate the test email submission flow.

**Comment by stuyam** [comment, marked as answer]
This is fixed. It was only an issue with the online editor and is fixed now 👍 thank you for bringing this to my attention!

**Reply by neillennon** [reply]
Hi @stuyam - I am facing an issue which i think is linked to this, whereby the inline editor renders a desktop and mobile view that is not the same as the test email sent on outlook (desktop) and Gmail ( desktop and mobile- iPhone 12 ). One specific issue I am facing is that the display on mobile in the inline editor is stacked vertically, however, when the actual email arrives it's still the "desktop" view but on mobile which looks terrible

**Reply by stuyam** [reply]
Can you post a HTML test of the example you are having issues with and what you are using to render it?
